### PR TITLE
Adds dev dep on envy for .env loading/parsing

### PR DESCRIPTION
**Pull request by robacarp**
Currently the docs say "source the .env file to add the variables to your shell". This makes that step unnecessary because envy does that automatically.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -762,8 +762,7 @@ $ docker-compose down
 If you'd like to test without docker you can do so by following the instructions below:
 
 1. Install dependencies with `$ crystal deps`
```
Comment (Blacksmoke16) [inline README.md]: This doesn't exist anymore.  Prob should update that as well.

**Comment by Blacksmoke16** [comment]
We were talking a bit about this in #180.  The issue is that as of now there isn't a better way to handle supplying the connection strings for running the tests for granite itself (since it needs all 3), which this change affects.  

That issue kinda ballooned out of scope, but i will be making a PR this weekend that will at least throw a warning if someone is using ENV variables for their app, as we want to deprecate reading ENV variables in favor of just doing `Granite.settings.database_url = xxx`.

**Comment by faustinoaq** [comment]
> That issue kinda ballooned out of scope, but i will be making a PR this weekend that will at least throw a warning if someone is using ENV variables/config file for their app, as we want to deprecate reading ENV variables/config files in favor of just doing Granite.settings.database_url = xxx.

@Blacksmoke16 So, What happens if I need to change my `database_url` on production, Do I need to recompile my app? :sweat_smile: 

Should I do this :point_down: then?

```cr
Granite.settings.database_url = ENV["DATABASE_URL"] if ENV["DATABASE_URL"]?
```

**Comment by Blacksmoke16** [comment]
Well the idea is that you can still use ENV variables like `Granite.settings.database_url = ENV["DATABASE_URL"]` as you have.  But it shouldn't be Granite's responsibility to look for config files or read ENV variables directly.

**Comment by eliasjpr** [comment]
@Blacksmoke16 I completely agree about it should not be Granite responsibility to look for environment I have raised this concern in the past. For that regard we should remove the `ENV["DATABASE_URL"]` from source code. :man_shrugging:

**Comment by drujensen** [comment]
Just to provide history and perspective on this.  

Granite was following the same standard that Rails uses for reading the DATABASE_URL environment variable if it exists.  I did this to make it easy to use but I am the first to admit that DATABASE_URL has bit me in the butt on several occasions in Rails projects where I wasn't aware the ENV variable was set and it would override the `config/databsase.yml`. 

I agreed that we want to remove this but I preferred deprecating it instead of breaking old sites (specifically mine) ;-)  I have moved or shutdown these sites so I'm good with removing it now.  :-D

**Comment by drujensen** [comment]
@robacarp Is this PR still valid?

**Comment by robacarp** [comment]
@drujensen Yes and no, but I think until the adapter thing is resolved it can be closed.

**Pull request closed without merging**